_posts/2020-01-01-2019-summary.md

```diff
@@ -0,0 +1,178 @@
+* [`all_match`](/docs/current/functions/array.html#all_match),
+  [`any_match`](/docs/current/functions/array.html#any_match) and 
+  [`none_match`](/docs/current/functions/array.html#none_match).
+* Support for higher-order [`reduce_agg`](/docs/current/functions/aggregate.html#reduce_agg) 
+  function in aggregation queries.
+* New datetime functions: [`last_day_of_month`](/docs/current/functions/datetime.html#last_day_of_month),
+  [`at_timezone`](/docs/current/functions/datetime.html#at_timezone) and 
+  [`with_timezone`](/docs/current/functions/datetime.html#with_timezone).
+
+## Geospatial
+
+* New geospatial functions: 
+  [`ST_Points`](/docs/current/functions/geospatial.html#ST_Points), 
+  [`ST_Length`](/docs/current/functions/geospatial.html#ST_Length), 
+  [`ST_Area`](/docs/current/functions/geospatial.html#ST_Area), 
+  [`line_interpolate_point`](/docs/current/functions/geospatial.html#line_interpolate_point) and 
+  [`line_interpolate_points`](/docs/current/functions/geospatial.html#line_interpolate_points).
+* `SphericalGeography` type and [related functions](/docs/current/functions/geospatial.html#to_spherical_geography) 
+  to support spatial features in geographic coordinates (latitude / longitude) on spherical model of the earth.
+* Support for Google Maps Polyline format via [`to_encoded_polyline`](/docs/current/functions/geospatial.html#to_encoded_polyline)
+  and [`from_encoded_polyline`](/docs/current/functions/geospatial.html#from_encoded_polyline) functions.
+* [`geometry_from_hadoop_shape`](/docs/current/functions/geospatial.html#geometry_from_hadoop_shape) to decode geometry objects in 
+  Spatial Framework for Hadoop representation
+
+## Security
+
+* [Role-based access control](/docs/current/sql/create-role.html) and related commands.
+* [INVOKER security mode](/docs/current/sql/create-view.html#security) for views, which allows views to be run using the permissions of the 
+  current user.
+* Prevent replay attacks and result hijacking in client APIs.
+* JWT-based [internal communication](/docs/current/security/internal-communication.html#internal-authentication) authentication.
+* Credential passthrough, which allows Presto to authenticate with the underlying data source with 
+  credentials provided by the user running a query.
+* Impersonation for [Hive metastore](/docs/current/connector/hive.html#hive-thrift-metastore-configuration-properties).
+* Support for reading and writing encrypted files in HDFS using Hadoop KMS.
+
+## Cloud Integration
+
+* Support for Azure Data Lake Blob and ADLS Gen2 storage.
+* Support for [Google Cloud Storage](/docs/current/connector/hive-gcs-tutorial.html).
+* Several [performance improvements](https://github.com/prestosql/presto/pull/142) for AWS S3.
+
+## CLI
+
+* Syntax highlighting and multi-line editing.
+ 
+  ![](/assets/blog/2019-review/presto-cli.jpeg)
+
+* JSON output format and improvements to CSV output format.
+
```

Comment:
```
* Support & stability improvements for running CLI with Java 11.
```

This also applies to JDBC, so maybe call the section "CLI & JDBC", so we don't add a new JDBC section with just one bullet

Reply:
What do the "allow committing empty transaction" and "restoring autocommit mode" help with? Was it better support for some third-party tool? I prefer to describe in terms of the impact.

Reply:
I didn't check with JDBC specs, but this makes it more compliant with other JDBC drivers and so it makes Presto JDBC more compatible with libraries.
(Specifically, i found this issues while trying to use Presto JDBC with a popular library)